### PR TITLE
CSL 2338 - Export catalog parameters as types

### PR DESCRIPTION
**Pull request by mocca102**

### CODE REVIEW

File: src/types/catalog.d.ts [hunk 1]
```diff
@@ -14,6 +14,236 @@ import {
...
+  ids?: string[];
+  itemId?: string;
+  numResultsPerPage?: number;
+  page?: number;
+}
+
+export interface AddItemGroupParameters {
+  id: string;
+  name: string;
+  parent_id?: string;
```
Comment (stanlp1) [inline src/types/catalog.d.ts:67]: I think for node, with the pr from 1 month ago, we changed all the parameters to camel case. Should we change the types as well?

**Review by stanlp1 — CHANGES_REQUESTED** [review]
Minor comment about casing. Rest looks good to me.

**Comment by stanlp1** [comment]
Screenshot 2023-05-17 at 5 32 47 PM


Also, should we use these exported types in the function types themselves? That way we can keep it in one place.

File: src/types/catalog.d.ts [hunk 2]
```diff
@@ -14,6 +14,236 @@ import {
...
 export default Catalog;
 
+export interface CreateOrReplaceItemsParameters {
+  items: Item[];
+  force?: boolean;
+  notificationEmail?: string;
+  section?: string;
+}
+
+export interface UpdateItemParameters extends CreateOrReplaceItemsParameters {}
```
Comment (stanlp1) [inline src/types/catalog.d.ts:24]: I think we are missing on_missing here too.

**Comment by mocca102** [comment]
@stanlp1 Created stories for the casing and missing parameters to be addressed in a separate PR
https://linear.app/constructor/issue/CSL-2414/node-sdk-add-on-missing-parameter-type-to-catalog-module
https://linear.app/constructor/issue/CSL-2412/node-sdk-update-parameters-casing-to-be-camelcase

**Pull request merged**